### PR TITLE
feat (api, cli): encrypt variable

### PR DESCRIPTION
**Pull request by fsamin**
To be able to write secret in the CDS yaml files, we can now encrypt data with the project GPG key with the cli

### CODE REVIEW

File: cli/cdsctl/encrypt.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import (
+	"fmt"
+
+	"github.com/ovh/cds/cli"
+)
+
+var (
+	encryptCmd = cli.Command{
+		Name:  "encrypt",
+		Short: "Encrypt variable into your CDS project",
```
Comment (bnjjj) [inline cli/cdsctl/encrypt.go]: missing long documentation for public doc (explain usecases for example)

File: sdk/cdsclient/client_project_variable.go [hunk 2]
```diff
@@ -36,3 +36,15 @@ func (c *client) ProjectVariableGet(projectKey string, varName string) (*sdk.Var
 	}
 	return variable, nil
 }
+
+func (c *client) VariableEnrypt(projectKey string, varName string, content string) (*sdk.Variable, error) {
```
Comment (bnjjj) [inline sdk/cdsclient/client_project_variable.go]: s/VariableEnrypt/VariableEncrypt

File: sdk/cdsclient/interface.go [hunk 3]
```diff
@@ -143,6 +143,7 @@ type ProjectVariablesClient interface {
 	ProjectVariableDelete(projectKey string, varName string) error
 	ProjectVariableGet(projectKey string, varName string) (*sdk.Variable, error)
 	ProjectVariableUpdate(projectKey string, variable *sdk.Variable) error
+	VariableEnrypt(projectKey string, varName string, content string) (*sdk.Variable, error)
```
Comment (bnjjj) [inline sdk/cdsclient/interface.go]: s/VariableEnrypt/VariableEncrypt

File: cli/cdsctl/encrypt.go (hunk 1 above)
Comment (yesnault) [reply cli/cdsctl/encrypt.go]:
```


Create a secret variable:


    $ cdsctl encrypt MYPROJECT my-data my-super-secret-value
    mydata: 01234567890987654321

The command returns the value: `01234567890987654321`. You can use this value in a configuration file.

Example of use case: Import an environment with a secret.

Create an environment file to import :


    $ cat << EOF > your-environment.json
    name: your-environment
    values:
    aReadableVariable:
        type: string
        value: registry.ovh.net/engine/http2kafka
    APasswordVariable:
        type: password
        value: 01234567890987654321
    EOF


Then, import then environment:

    cds environment import MYPROJECT your-environment.json



```

File: cli/cdsctl/encrypt.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+		Long:  "To be able to write secret in the CDS yaml files, you have encrypt data with the project GPG key",
+		Example: `cdsctl encrypt MYPROJECT my-data my-super-secret-value
+mydata: 01234567890987654321`,
```
Comment (yesnault) [inline cli/cdsctl/encrypt.go]: my-data

**Pull request merged**